llama_stack/providers/tests/agents/test_agents.py

```diff
@@ -272,6 +272,70 @@ async def test_create_agent_turn_with_brave_search(
 
         check_turn_complete_event(turn_response, session_id, search_query_messages)
 
+    @pytest.mark.asyncio
```

Comment:
can you combine the implementation of the rest of the test with that of brave_search? it's a large amount of duplication and is certain to get out of sync

Reply:
Combined the duplicated codes in `create_agent_turn_with_search_tool`.